rules/service.go

```diff
@@ -0,0 +1,473 @@
+
+	// ErrMalformedEntity indicates malformed entity specification (e.g.
+	// invalid username or password).
+	ErrMalformedEntity = errors.New("malformed entity specification")
+
+	// ErrUnauthorizedAccess indicates missing or invalid credentials provided
+	// when accessing a protected resource.
+	ErrUnauthorizedAccess = errors.New("missing or invalid credentials provided")
+
+	// ErrNotFound indicates a non-existent entity request.
+	ErrNotFound = errors.New("non-existent entity")
+
+	// ErrKuiperServer indicates internal kuiper rules engine server error
+	ErrKuiperServer = errors.New("kuiper internal server error")
+)
+
+// Info is used to fetch kuiper running instance data
+type Info struct {
+	Version       string `json:"version"`
+	Os            string `json:"os"`
+	UpTimeSeconds int    `json:"upTimeSeconds"`
+}
+
+// Service specifies an API that must be fullfiled by the domain service
+// implementation, and all of its decorators (e.g. logging & metrics).
+type Service interface {
+	Info(ctx context.Context) (Info, error)
+	CreateStream(ctx context.Context, token string, stream Stream) (string, error)
+	UpdateStream(ctx context.Context, token string, stream Stream) (string, error)
+	ListStreams(ctx context.Context, token string) ([]string, error)
+	ViewStream(ctx context.Context, token, name string) (StreamInfo, error)
+	Delete(ctx context.Context, token, name, kind string) (string, error)
+
+	CreateRule(ctx context.Context, token string, rule Rule) (string, error)
+	UpdateRule(ctx context.Context, token string, rule Rule) (string, error)
+	ListRules(ctx context.Context, token string) ([]RuleInfo, error)
+	ViewRule(ctx context.Context, token, name string) (Rule, error)
+	GetRuleStatus(ctx context.Context, token, name string) (map[string]interface{}, error)
+	ControlRule(ctx context.Context, token, name, action string) (string, error)
+}
+
+type reService struct {
+	kuiperURL string
+	auth      mainflux.AuthServiceClient
+	sdk       SDK.SDK
+	logger    logger.Logger
+}
+
+var _ Service = (*reService)(nil)
+
+// New instantiates the re service implementation.
+func New(url string, auth mainflux.AuthServiceClient, sdk SDK.SDK, logger logger.Logger) Service {
+	return &reService{
+		kuiperURL: url,
+		auth:      auth,
+		sdk:       sdk,
+		logger:    logger,
+	}
+}
+
+func (re *reService) Info(_ context.Context) (Info, error) {
+	res, err := http.Get(re.kuiperURL)
+	if err != nil {
+		return Info{}, errors.Wrap(ErrKuiperServer, err)
+
+	}
+	defer res.Body.Close()
+
+	var i Info
+	err = json.NewDecoder(res.Body).Decode(&i)
+	if err != nil {
+		return Info{}, errors.Wrap(ErrKuiperServer, err)
+	}
+
+	return i, nil
+}
+
+func (re *reService) CreateStream(ctx context.Context, token string, stream Stream) (string, error) {
+	ui, err := re.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return "", ErrUnauthorizedAccess
+	}
+	_, err = re.sdk.Channel(stream.Topic, token)
```

Comment:
this doesn't seem right to me, maybe I dont get it but it seems that 
you want to check if the access is allowed through sdk but this should be done through the gRPC  call
similar to the Identify call above,and here you are making http request